src/lkl/setup.c

```diff
@@ -324,22 +320,39 @@ static void lkl_mount_overlayfs(
 {
     char opts[200];
     snprintf(
-            opts,
-            sizeof(opts),
-            "lowerdir=%s,upperdir=%s,workdir=%s",
-            lower_dir, upper_dir, work_dir);
+        opts,
+        sizeof(opts),
+        "lowerdir=%s,upperdir=%s,workdir=%s",
+        lower_dir,
+        upper_dir,
+        work_dir);
     int err = lkl_sys_mount("overlay", (char*)mnt_point, "overlay", 0, opts);
     if (err != 0)
     {
         sgxlkl_fail("lkl_sys_mount(overlayfs): %s\n", lkl_strerror(err));
     }
 }
 
+typedef struct
+{
+    bool create;
+    const char* destination;
+    size_t key_len;
+    uint8_t* key;
+    const char* key_id;
+    bool fresh_key;
+    bool readonly;
+    const char* roothash;
+    size_t roothash_offset;
+    size_t size;
+    bool overlay;
+} disk_config_t;
```

Comment:
Should this live here?

Reply:
"here": yes. "live": no. This is the way the rest of the code expects the disk state to be organized and I didn't feel like rewriting all of the mounting and setup logic. The old code just abused the config structures for this, so now it's at least separated from the config.

Reply:
Maybe to resolve this it would be good to understand what the difference is between this struct and what's coming from the config.

Reply:
ping @wintersteiger

Reply:
I'll take another look, but this can easily be done later and by anybody. The difference is that root and mount options are merged in one disk config. To clean this up, we would have to separate root and mount portions of each function, which is clearly the right thing to do, but it's not necessary immediately.